src/coreclr/pal/src/arch/arm64/context2.S

```diff
@@ -104,6 +105,32 @@ LOCAL_LABEL(Done_CONTEXT_INTEGER):
     sub x0, x0,   CONTEXT_FLOAT_CONTROL_OFFSET + CONTEXT_NEON_OFFSET
 
 LOCAL_LABEL(Done_CONTEXT_FLOATING_POINT):
+    ldr x1, [x0, CONTEXT_XSTATEFEATURESMASK_OFFSET]
```

Comment:
It should check the CONTEXT_XSTATE in CONTEXT_ContextFlags first and check the features mask only if the CONTEXT_XSTATE is set.